common.mk

```diff
@@ -3,6 +3,13 @@
 #########################################################################################
 SHELL=/bin/bash
 
+#########################################################################################
+# specific make variables/rules from subprojects
+#########################################################################################
+include $(base_dir)/generators/ariane/ariane.mk
+include $(base_dir)/generators/tracegen/tracegen.mk
+include $(base_dir)/tools/dromajo/dromajo.mk
+
```

Comment:
I get that making this change against the dromajo PR is done to show why this splitting is useful, but I'm still apprehensive about making so many changes in a chipyard feature branch. If its not too much work, and people like these changes, can we start with a PR into the dev branch that does this restructuring for just ariane and tracegen, and merge in dromajo support later?

Reply:
Sure, we can split this up... I just want to show how this might be done nicely with something that needs it more (i.e. Dromajo). If people agree on this approach then we can migrate it to `dev`.

Reply:
so what happens when these projects settings conflict?

Reply:
this is strictly better than what we had before though.

Reply:
Currently, everything is additive (see the `+=` for all the flags in their respective `mk`s). However, I haven't put much thought into how we deal with conflicts (for ex. when `+drj_bin=` is defined twice).

Reply:
I know that the simulator will probably error... but maybe having some sort of check in `common.mk` would be nice? Maybe warn if full words match between `mk`s for the SIM_FLAGS variable (i.e. so that `+arg +arg` is caught)?